### PR TITLE
Initial Version - RDS EOL

### PR DESCRIPTION
**Pull request by chassing**
Introduce the first script to parse the RDS release calendars.

Create a dev env:

```bash
$ # create and activate a venv with python 3.11
$ pip install poetry
$ poetry install
```

Run it locally in your dev env:
```bash
$ # create and activate a venv with python 3.11
$ python -m aws_generated_data --debug rds-eol fetch --engines postgres:https://docs.aws.amazon.com/AmazonRDS/latest/PostgreSQLReleaseNotes/postgresql-release-calendar.html --engines mysql:https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/MySQL.Concepts.VersionMgmt.html --output foo.yml
```

### CODE REVIEW

File: aws_generated_data/commands/rds_eol.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+def parse_date(date_str: str) -> datetime:
+    try:
+        return datetime.strptime(date_str, "%B %Y")
+    except ValueError:
+        return datetime.strptime(date_str, "%d %B %Y")
+
+
+def parse_aws_release_calendar(page: str) -> list[CalItem]:
+    items: list[CalItem] = []
+    soup = BeautifulSoup(page, "html.parser")
```
Comment (hemslo) [inline aws_generated_data/commands/rds_eol.py:73]: `html5lib` maybe a better one [doc](https://tedboy.github.io/bs4_doc/9_specifying_the_parser.html)

File: aws_generated_data/commands/rds_eol.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    version_table = soup.find_all("table")[0]
```
Comment (hemslo) [inline aws_generated_data/commands/rds_eol.py:74]: Find first `table` seems a little hard coded, but I can't predicate how that page may change in the future, so good for now.

Btw, to find first, can use `soup.find("table")` directly.

File: aws_generated_data/commands/rds_eol.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    rds_items = read_output_file(output)
+    for engine in engines:
+        log.info(f"Processing {engine} ...")
+        rds_items += get_rds_eol_data(engine)
+
+    rds_items = filter_rds_items(
+        rds_items,
+        expired_date=date.today() - timedelta(days=clean_up_days),
+    )
+    write_output_file(output, sorted(set(rds_items), reverse=True))
```
Comment (hemslo) [inline aws_generated_data/commands/rds_eol.py:150]: What if some `eol` data changed from last time? Will both old eol and new eol saved to output file?

**Review by hemslo — COMMENTED** [review]
Nice new setup!

File: aws_generated_data/commands/rds_eol.py (hunk 1 above)
Comment (chassing) [reply aws_generated_data/commands/rds_eol.py]: Change in [e59f03a](https://github.com/app-sre/aws-generated-data/pull/1/commits/e59f03aea01bcf805bc5ae5e6f0e95618e90f293)

File: aws_generated_data/commands/rds_eol.py (hunk 2 above)
Comment (chassing) [reply aws_generated_data/commands/rds_eol.py]: Changed in [882c573](https://github.com/app-sre/aws-generated-data/pull/1/commits/882c573a52150cf134b79a12043780309542da21)

File: aws_generated_data/commands/rds_eol.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+    def __lt__(self, other: Any) -> bool:
+        if not isinstance(other, RdsItem):
+            return False
+        return (self.engine, self.version) < (other.engine, other.version)
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, RdsItem):
+            return False
+        return (self.engine, self.version) == (other.engine, other.version)
```
Comment (hemslo) [inline aws_generated_data/commands/rds_eol.py:39]: partial override `__eq__` require carefully checking, as `rds_item_a == rds_item_b` is `True` but they may have different `eol`, how about convert `rds_items` to dict for processing? Like this:

```python
    rds_items_dict = {
        (item.engine, item.version): item
        for item in rds_items
    }

    for engine in engines:
        log.info(f"Processing {engine} ...")
        for item in get_rds_eol_data(engine):
            rds_items_dict[(item.engine, item.version)] = item

    rds_items = filter_rds_items(
        rds_items_dict.values(),
        expired_date=date.today() - timedelta(days=clean_up_days),
    )
```

**Review by hemslo — APPROVED** [review]
LGTM! Just a small refactor suggestion for updating items.

File: aws_generated_data/commands/rds_eol.py (hunk 4 above)
Comment (chassing) [reply aws_generated_data/commands/rds_eol.py]: Yeah, this is easier to read! 🙇 
Applied in [2dfceff](https://github.com/app-sre/aws-generated-data/pull/1/commits/2dfceff27779c1b51d379f4315e19aebf5dada0c)

**Pull request merged**